### PR TITLE
Complete builder error handling

### LINKED ISSUE #677: Builder Does Not Respond to Service Errors
When the Learning Object Service responds with a 500 error, the builder does not communicate the failure to the user. From the user perspective, everything is working and they should keep going. This should be fixed to better communicate what is going on to our users.

### PR DESCRIPTION
**Pull request by gshaw1997**
This PR completes fix to issue where the builder did not respond to service errors by adding 
- Modal that navigates user to dashboard when multiple 500 level errors occur successively
- Handler for builder errors that trigger toast messages.
- Hiding of service interaction badge on error

**Notable UI Changes**
- Error Modal

screenshot from 2019-03-04 18-36-18

Open to suggestions on language in modal!
 
Closes #677

### CODE REVIEW

File: src/app/onion/learning-object-builder/learning-object-builder.component.ts [hunk 1]
```diff
@@ -100,10 +100,14 @@ export class LearningObjectBuilderComponent implements OnInit, OnDestroy {
 
   adminMode: boolean;
 
+  showServiceFailureModal = false;
+  adminDashboardURL = 'https://admin.clark.center';
```
Comment (nvisal1) [inline src/app/onion/learning-object-builder/learning-object-builder.component.ts]: Why are we hardcoding a URL here?

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 2]
```diff
@@ -933,8 +933,8 @@ export class BuilderStore {
     error: HttpErrorResponse,
     builderError: BUILDER_ERRORS
   ) {
-    this.serviceInteraction$.next(false);
-    if (error.status === 500) {
+    this.serviceInteraction$.next(null);
+    if (error.status === 0 || error.status === 500) {
```
Comment (nrwinner) [inline src/app/onion/learning-object-builder/builder-store.service.ts]: Can we have a comment here to explain the usage of `error.status === 0`?

File: src/app/onion/learning-object-builder/learning-object-builder.component.ts (hunk 1 above)
Comment (gshaw1997) [reply src/app/onion/learning-object-builder/learning-object-builder.component.ts]: Meant to go back and update. Nice catch!

File: src/app/onion/learning-object-builder/learning-object-builder.component.html [hunk 3]
```diff
@@ -28,4 +28,16 @@
...
     {{ errorMessage }}
   </span>
-</ng-template>
\ No newline at end of file
+</ng-template>
+
+<!-- Service Error Popup -->
+<clark-popup *ngIf="showServiceFailureModal" (closed)="routeToDashboard()">
+  <div #popupInner class="popup-inner">
+    <div class="modal-title">Network error</div>
```
Comment (DonSeannelly) [inline src/app/onion/learning-object-builder/learning-object-builder.component.html]: Can we override this style's `color` field to use the new blue?

File: src/app/onion/learning-object-builder/learning-object-builder.component.html [hunk 4]
```diff
@@ -28,4 +28,16 @@
...
+    <div class="modal-text">We are currently unable to save your work due to a network related issue. Please check back
+      later and try again.</div>
+    <div class="btn-group center">
+      <button class="button good" (click)="routeToDashboard()">Back to Dashboard</button>
```
Comment (DonSeannelly) [inline src/app/onion/learning-object-builder/learning-object-builder.component.html]: Can we override this style's `background` field to use the new blue?

**Comment by gshaw1997** [comment]
@DonSeannelly Updated blue
Screenshot from 2019-03-11 16-04-15

**Pull request merged**